lib/public/UrlChecker.js

```diff
@@ -115,4 +122,33 @@ UrlChecker.prototype.resume = function()
 
 
 
+UrlChecker.prototype.clearCache = function()
+{
+	this.cache = {};
+};
+
+
+
+//::: PRIVATE FUNCTIONS
+
+
+
+function shouldQueryUrl(url, context)
+{
+	if (context.options.cacheResponses === true)
+	{
+		if (context.cache[url] === true)
+		{
+			return false;
```

Comment:
Because of this, the CLI will indefinitely pause: https://github.com/stevenvachon/broken-link-checker/blob/master/lib/cli.js#L149-L152

For example:
```html
<a href="/page.html">link1</a>
<a href="/page.html">link2</a>
<a href="/page2.html">link3</a>
```
will produce three link objects, each with their own index (0,1,2).

The CLI logger preserves HTML link order using those indexes and will wait for index 1 (link2) but will never get it.